en/middleware/micrortps.md

```diff
@@ -0,0 +1,271 @@
+- The *Agent* and any *FastRTPS* applications are connected via UDP, and may be on the same or another device. In a typical configuration they will both be on the same system (e.g. a development computer, Linux companion computer or compute board), connected to the *Client* over a Wifi link or via USB.
+
+
+## Code generation
+
+The code needed for the *Client*, *Agent*, *CDR serialization/deserialization* of uORB messages, and the definition of the associated RTPS messages is automatically generated when the PX4 Firmware is compiled (and can also be [generated manually](micrortps_manual_code_generation.md)). 
+
+> **Note** [Fast RTPS must be installed](../setup/fast-rtps-installation.md) in order to generate the required code! 
+
+The source code for the *Client* and *Agent* will be generated in **build\__BUILDPLATFORM_/src/modules/micrortps_bridge**, under the **/micrortps_client** and **/micrortps_agent** folders (respectively).
+
+
+### Supported uORB/RTSP messages
+
+The set of uORB topics that can be sent and received by the bridge are listed in the **.cmake** file (**cmake/configs**) for each target platform:
+
+```cmake
+set(config_rtps_send_topics
+  sensor_combined
+   # Add new topic...
+   )
+
+set(config_rtps_receive_topics
+   vehicle_command
+   # Add new topic...
+   )
+```
+
+A uORB serialization function will be generated for each topic in both lists. An IDL file will also be created for each topic from the associated uORB **.msg** files 
+
+> **Note** FastRTSP uses IDL files to define the structure of RTPS messages (in this case, RTPS messages for sending uORB topics). The IDL files are compiled to C++ by the *fastrtpsgen* tool, which is then used by the Agent and other FastRTPS applications.
+
+
+#### uORB serialization code
+
+Serialization functions are generated for the uORB topics that are to be sent/received. 
+For example, the following functions would be generated for the *sensor_combined.msg*:
+
+```sh
+void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+```
+
+
+## Client (PX4 Firmware)
+
+The *Client* source code is generated, and the executable is automatically built and included when you make the firmware (in the normal way). For example, to build and upload the firmware:
+
+For NuttX/Pixhawk flight controllers:
+```sh
+make px4fmu-v4_default upload
+```
+For Qualcomm Snapdragon Flight:
+```sh
+$ make eagle_default upload
+  ```
+
+After uploading the firmware, the *Client* application can be launched by typing its name and passing a variable number of arguments as shown below:
```

Comment:
@vilhjalmur89 @LorenzMeier Manually launching the RTPS client sounds very wrong. Normally a process in firmware that is needed should be started by the configuration and then run forever. What is going on here?